addon/components/yeti-table/thead.js

```diff
@@ -19,19 +18,11 @@ import Component from '@ember/component';
   @yield {object} head
   @yield {Component} head.row
 */
-@tagName('')
+// eslint-disable-next-line ember/no-empty-glimmer-component-classes
 class THead extends Component {
-  theme;
-
-  sortable;
-
-  sortSequence;
-
-  parent;
-
-  columns;
-
-  onColumnClick;
+  /**
+   * @argument {function} onColumnClick - action that is called when the column header is clicked
+   */
```

Comment:
I believe the format should be something like this:

```js
    /**
     * String to be shown in the attribution control, e.g. "© OpenStreetMap contributors". It describes the layer
     * data and is often a legal obligation towards copyright holders and tile providers.
     *
     * @argument attribution
     * @type {String}
     */
```

I believe the standard is to capitalize all js types, in this case `Function`.

Reply:
Since the docs for all params https://jsdoc.app/tags-param.html have the type included as I have done, not sure why you would use @type separately.    Looking at the type docs I would only use it for "Type Definitions" (the last block) look at the examples in the blocks "Variable number of that type" and "Optional parameter", even though this is the type docs, they aren't using type but the variation of param I am using.


You are correct though, should have been capped.


Will update once you let me know if this variation is fine or if you want me to use @type

Reply:
I was just following what ember-cli-addon-docs recommends in their documentation. I have no idea if your version results in the same generated api docs website. But I do know their version results in a complete argument definition like this:

![image](https://user-images.githubusercontent.com/631691/107539859-4c6af100-6bbd-11eb-924e-7ed92f67e271.png)

Here is an example component which has documented arguments: https://github.com/ember-learn/ember-cli-addon-docs/blob/master/sandbox/app/components/yuidoc-component.js

It is also my personal preference to have things visually separated out, but I understand this isn't a strong "argument". 🙃